### ISSUE TITLE
CANNOT_DISABLE_PIT_WITH_BACKUP_COMPLIANCE_POLICY 

### ISSUE THREAD

**Issue by mvaduguru87** [mongodb/terraform-provider-mongodbatlas#1855]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
 Terraform Version 1.6.4
Terraform MongoDB Atlas Provider Version 1.14.0
```
###Terraform Edition
```hcl
Terraform OSS
```

### Terraform Configuration File 
```hcl
#resource "mongodbatlas_backup_compliance_policy" "this" {
  count                      = var.enabled ? 1 : 0
  project_id                 = var.project_id
  authorized_email           = ###
  copy_protection_enabled    = false
  pit_enabled                = true
  encryption_at_rest_enabled = false
  authorized_user_first_name = "###"
  authorized_user_last_name = "###"
restore_window_days = var.restore_window_days
...
}
```

### Steps to Reproduce

```hcl
Plan/Apply 
```

### Actual Behavior

When the flag  enabled is set to true , the plan is getting generated but when applying the plugin is crashing. 
[plugincrash.log](https://github.com/mongodb/terraform-provider-mongodbatlas/files/13982324/plugincrash.log)

When the flag enabled is set to false , the plan is getting generated but when applying getting the error 
Error: error creating MongoDB Cluster: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/65a8075d15a5527895928ac4/clusters: 400 (request "CANNOT_DISABLE_PIT_WITH_BACKUP_COMPLIANCE_POLICY") Continuous Cloud Backup is required by Backup Compliance Policy.
![Screenshot 2024-01-18 at 5 43 42 PM](https://github.com/mongodb/terraform-provider-mongodbatlas/assets/139978544/fa52abc0-42e5-4062-a6f7-4ce2d7760000)

-->

### References
(https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/resources/backup_compliance_policy)

**Comment by maastha** [comment]
Hi @mvaduguru87 
Thank you for creating this issue, I could not reproduce this and was able to successfully update `pit_enabled` to true.

Could you please help us with below details to look further into this issue?
- Terraform configuration file used to reproduce the issue (including cluster configuration and complete config for mongodbatlas_backup_compliance_policy)
- Terraform log files from the run where the issue occurred (the provided log file only shows `panic: interface conversion: interface {} is nil, not string`, not the errors you got from the API, you could try `export TF_LOG=”TRACE”` as mentioned in our guidelines)


I would also request you to provide further details to help with  [one-click reproducible issues principle](https://github.com/mongodb/README.md#one-click-reproducible-issues-principle) as mentioned in our guidelines.

**Comment by mvaduguru87** [comment]
Certainly!. We are using modules to create the clusters. Hence I am providing you with the output of the plan that still provides you cluster configuration.  Please find attached file terraformplan_1_14_0.txt for the details. 
[terraformplan_1_14_0.txt](https://github.com/mongodb/terraform-provider-mongodbatlas/files/13995612/terraformplan_1_14_0.txt)

I am also attaching the trace logs.  terraform_1.14_1_trace.log 
[terraform_1_14_1_trace.log](https://github.com/mongodb/terraform-provider-mongodbatlas/files/13995566/terraform_1_14_1_trace.log)


I also want to mention that since all our clusters that have already been created are using V1.13.1 mongo db provider version, I tried to use the same to run the clusters. I received the same error. I am attaching the configurations in file and tracelogs. 
[terraformplan_1_13_1.txt](https://github.com/mongodb/terraform-provider-mongodbatlas/files/13995607/terraformplan_1_13_1.txt)
[terraform_1_13_1__v1_trace.log](https://github.com/mongodb/terraform-provider-mongodbatlas/files/13995608/terraform_1_13_1__v1_trace.log)

The error in all these cases is same -  Error: error creating MongoDB Cluster: POST https://cloud.mongodb.com/api/atlas/v1.0/groups/65a8075d15a5527895928ac4/clusters: 400 (request "CANNOT_DISABLE_PIT_WITH_BACKUP_COMPLIANCE_POLICY") Continuous Cloud Backup is required by Backup Compliance Policy.

Please let me know if you need any more details.  Unfortunately I couldn't provide you a one -click reproducible files. I hope what is provided will help to identity the issue. 

Also please note that I removed all the sensitive information from the trace files.

**Comment by maastha** [comment]
Hi @mvaduguru87 

I was able to reproduce `panic: interface conversion: interface {} is nil, not string` error in v1.14.0. This is already fixed in https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1803 ([see comment](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1803/files#r1440240057)) and would be available in our next release planned in the next few weeks.


As for `CANNOT_DISABLE_PIT_WITH_BACKUP_COMPLIANCE_POLICY` error when creating the cluster, I'd request you to open a HELP support ticket https://support.mongodb.com/ to get support from the appropriate team as this seems like an issue with the way backup  compliance policy and the cluster are configured.

Note that I am not sure of the exact order in which updates were performed, but with the fix mentioned above I was able to successfully create a cluster with same configuration as yours (with `mongodbatlas_backup_compliance_policy.pit_enabled = true`) but received below error if I tried to update it to `false`

```
Error: error updating a Backup Compliance Policy: xxxxx: PUT https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/xxxxxx/backupCompliancePolicy: 400 (request "BACKUP_COMPLIANCE_POLICY_SETTINGS_INVALID") Invalid Backup Compliance Policy settings: a support ticket is required to turn off point-in-time option or decrease the restore window.
```
